test/extensions/http/early_header_mutation/header_mutation/header_mutation_test.cc

```diff
@@ -59,94 +48,27 @@ TEST(HeaderMutationTest, Basic) {
   HeaderMutation mutation(proto_mutation);
   NiceMock<Envoy::StreamInfo::MockStreamInfo> stream_info;
 
-  {
```

Comment:
Why get rid of all these tests? I realize we want to have this newer filter subsume many of the cases of the early header mutation filter, but not sure why the selective removal of tests in the current PR.

Reply:
In fact, they ware not removed but move to the https://github.com/envoyproxy/envoy/pull/25658/files#diff-e87c3085497f9392ed99f92ff4d3189cd16810be1574517625f72a08d75b6d7c.

We extracted some common code to the `common/http/header_mutation`, so all these old tests are moved to the common part.